components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/TenantSyncListener.java

```diff
@@ -0,0 +1,401 @@
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+                String domain = realmService.getTenantManager().getTenant(tenantId).getDomain();
+
+                tenantInfo.setTenantDomain(domain);
+                tenantInfo.setTenantId(tenantId);
+                tenantInfo.setActive(false);
+                sendEvent(tenantInfo, TenantManagement.ACTION_DEACTIVATE, TenantManagement.EVENT_ACTIVATE_TENANT_URI);
+            } catch (UserStoreException e) {
+                LOG.error("Error while deactivating tenant ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public void onSubscriptionPlanChange(int tenentId, String oldPlan, String newPlan) throws StratosException {
+        // Not implemented.
+    }
+
+    @Override
+    public int getListenerOrder() {
+
+        return 0;
+    }
+
+    @Override
+    public void onPreDelete(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    /**
+     * Check whether tenant sharing is enabled and whether the creation is only a root organization.
+     *
+     * @param tenantId id of the tenant
+     * @return boolean Tenant creation firing is enable or not.
+     */
+    private boolean canCreateTenant(int tenantId) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking if tenant can be created. Tenant ID: " + tenantId);
+        }
+
+        boolean canCreateTenant = false;
+
+        IdentityEventListenerConfig identityEventListenerConfig = IdentityUtil.readEventListenerProperty(
+                UserOperationEventListener.class.getName(), TenantSyncListener.class.getName());
+
+        if (StringUtils.isNotBlank(identityEventListenerConfig.getEnable())
+                && Boolean.parseBoolean(identityEventListenerConfig.getEnable())) {
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+
+                Tenant tenant = realmService.getTenantManager().getTenant(tenantId);
+                String organizationID = tenant.getAssociatedOrganizationUUID();
+
+                // check if the Organization Depth in the Hierarchy is 0. only then create the root org.
+                if (organizationID == null || IdentityMgtServiceDataHolder.getInstance().getOrganizationManager()
+                        .getOrganizationDepthInHierarchy(organizationID) == 0) {
+
+                    canCreateTenant = true;
+
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Tenant is a root organization and can be created. Tenant ID: "
```

Comment:
This is just string concatenation, so isDebugEnabled check is not needed.
It needed if we do any heavy operations or string formattings.

Reply:
I think this check is still necessary. Since string concatenation occurs before the debug() method is executed, it can be an expensive task. You can find more details in the article here: https://dzone.com/articles/measuring-string-concatenation